addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,25 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        status = {key: value for key, value in request.session.get('pending_coupon', {}).items() if key == 'promo_code'}
```

Comment:
:roll_eyes:

Reply:
There were more keys before, that's why... What do you think about:
```
        status = request.session.get('pending_coupon', {})
        status = {'promo_code': status['promo_code']} if status['promo_code'] else {}
```